arrow-fx/src/main/kotlin/io/arrowkt/frdomain/app/app.kt

```diff
@@ -0,0 +1,116 @@
+package io.arrowkt.frdomain.app
+
+import arrow.core.either
+import io.arrowkt.frdomain.Amount
+import io.arrowkt.frdomain.repository.interpreter.AccountInMemoryRepository
+import io.arrowkt.frdomain.service.AccountServiceException
+import io.arrowkt.frdomain.service.AccountType.CHECKING
+import io.arrowkt.frdomain.service.AccountType.SAVINGS
+import io.arrowkt.frdomain.service.interpreter.AccountService.credit
+import io.arrowkt.frdomain.service.interpreter.AccountService.debit
+import io.arrowkt.frdomain.service.interpreter.AccountService.open
+import io.arrowkt.frdomain.service.interpreter.ReportingService.balanceByAccount
+import java.math.BigDecimal
+
+suspend fun main() {
+    usecase1()
+    usecase2()
+    usecase3()
+    usecase4()
+}
+
+suspend fun usecase1() {
```

Comment:
Nit: these functions could be constrained to `AccountRepository` in their receiver.
`suspend fun AccountRepository.usecaseX(): Unit`

That way there is no need to wrap the inner code in, `with(AccountInMemoryRepository())` but leaves it upon to the `suspend fun main()` block.
Where you instead would call.

```kotlin
suspend fun main() = with(AccountInMemoryRepository()) {
    usecase1() // <-- reader syntax
    usecase2()
    usecase3()
    usecase4()}
```

That has the benefit of requiring less nesting overall in your code, and it leaves the actual implementation of the use-cases independent of the implementations by constraint.

Reply:
Thanks for the feedback. Fixed, with a slight change (because each use case needs an empty repo).